File: lab/python/textos/remover_espacos_exagerados.py
def remover_espacos_exagerados1(text, one_pharase=False):
    text = text.strip()

    if one_pharase:
        return text.replace(" ", "")

    uniao = ""
    cont = 0

    for line in text:
        if not line.isspace():
            uniao += line
            cont = 0
        else:
            if cont == 0:
                uniao += line
                cont += 1

    return uniao


def remover_espacos_exagerados2(text, one_pharase=False):
    text = text.strip()
    if one_pharase:
        return text.replace(" ", "")
    return " ".join(text.split())


texto = "esse  texto  veio todo quebrado"
print(remover_espacos_exagerados1(texto))
print(remover_espacos_exagerados2(texto))

texto = "o la"
print(remover_espacos_exagerados1(texto, one_pharase=True))
print(remover_espacos_exagerados2(texto, one_pharase=True))

File: lab/c/textos/remover_espacos_exagerados.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Função 1 — remove espaços exagerados (mantém um por vez)
void remover_espacos_exagerados1(const char *text, int one_pharase, char *saida) {
    int len = strlen(text);
    int cont = 0;
    int j = 0;

    // Ignora espaços no início e no fim
    int start = 0, end = len - 1;
    while (isspace((unsigned char)text[start])) start++;
    while (end >= 0 && isspace((unsigned char)text[end])) end--;

    for (int i = start; i <= end; i++) {
        char c = text[i];
        if (!isspace((unsigned char)c)) {
            saida[j++] = c;
            cont = 0;
        } else {
            if (one_pharase)
                continue; // remove todos os espaços
            if (cont == 0) {
                saida[j++] = ' '; // mantém só um espaço
                cont++;
            }
        }
    }
    saida[j] = '\0';
}

// Função 2 — versão "curta"
void remover_espacos_exagerados2(const char *text, int one_pharase, char *saida) {
    char temp[200];
    int i = 0, j = 0, cont = 0;
    int len = strlen(text);

    // Remove espaços no início e fim
    int start = 0, end = len - 1;
    while (isspace((unsigned char)text[start])) start++;
    while (end >= 0 && isspace((unsigned char)text[end])) end--;

    for (i = start; i <= end; i++) {
        if (isspace((unsigned char)text[i])) {
            if (one_pharase)
                continue;
            if (cont == 0) {
                temp[j++] = ' ';
                cont++;
            }
        } else {
            temp[j++] = text[i];
            cont = 0;
        }
    }
    temp[j] = '\0';
    strcpy(saida, temp);
}

// -------------------- Testes --------------------
int main(void) {
    char texto1[] = "esse  texto  veio todo quebrado";
    char saida[200];

    remover_espacos_exagerados1(texto1, 0, saida);
    printf("%s\n", saida);

    remover_espacos_exagerados2(texto1, 0, saida);
    printf("%s\n", saida);

    char texto2[] = "o la";
    remover_espacos_exagerados1(texto2, 1, saida);
    printf("%s\n", saida);

    remover_espacos_exagerados2(texto2, 1, saida);
    printf("%s\n", saida);

    return 0;
}
